suites/apis/googleServiceAccountKeyTest.js

```diff
@@ -0,0 +1,349 @@
+  ).to.equal(0);
+
+  // Get temporary google creds
+  let tempCredsRes = await fence.complete.createTempGoogleCreds(users.user0.accessTokenHeader);
+  const keyId1 = tempCredsRes.body.private_key_id;
+  console.log(`Generated key ${keyId1}`);
+
+  // Get temporary google creds with custom expiration
+  tempCredsRes = await fence.complete.createTempGoogleCreds(users.user0.accessTokenHeader, EXPIRES_IN);
+  const keyId2 = tempCredsRes.body.private_key_id;
+  console.log(`Generated key ${keyId2}`);
+
+  // Get list of current creds
+  getCredsRes = await fence.do.getUserGoogleCreds(users.user0.accessTokenHeader);
+  let credsList2 = getCredsRes.access_keys;
+  console.log('Current SA creds:');
+  console.log(credsList2);
+
+  // Delete a key
+  await fence.do.deleteTempGoogleCreds(
+    keyId1, users.user0.accessTokenHeader);
+
+  // Get list of current creds
+  getCredsRes = await fence.do.getUserGoogleCreds(users.user0.accessTokenHeader);
+  let credsList3 = getCredsRes.access_keys;
+
+  // Clean up
+  console.log('cleaning up');
+
+  await fence.do.deleteTempGoogleCreds(
+    keyId2, users.user0.accessTokenHeader);
+
+  await fence.do.deleteGoogleServiceAccount(
+    users.user0,
+    googleProject.serviceAccountEmail,
+  );
+
+  await fence.do.unlinkGoogleAcct(
+    users.user0,
+  );
+
+  // Asserts
+  expect(credsList2.length,
+    'The 2 generated SA keys should be listed'
+  ).to.equal(2);
+
+  let key1 = credsList2.filter(key => key.name.includes(keyId1));
+  expect(key1.length,
+    'The generated SA key should be listed'
+  ).to.equal(1);
+
+  let start = Date.parse(key1[0].validAfterTime);
+  let end = Date.parse(key1[0].validBeforeTime);
+  expect(
+    (end - start) / 10000,
+    `The key should be set to expire in ${fence.props.linkExtendDefaultAmount} secs`
+  ).to.be.within(
+    fence.props.linkExtendDefaultAmount - 5,
+    fence.props.linkExtendDefaultAmount + 5
+  );
+
+  let key2 = credsList2.filter(key => key.name.includes(keyId2));
+  expect(key2.length,
+    'The generated SA key should be listed'
+  ).to.equal(1);
+
+  start = Date.parse(key2[0].validAfterTime);
+  end = Date.parse(key2[0].validBeforeTime);
+  expect(
+    (end - start) / 10000,
+    `The key should be set to expire in ${EXPIRES_IN} secs`
+  ).to.be.within(EXPIRES_IN - 5, EXPIRES_IN + 5);
+
+  expect(credsList3.length,
+    'Only 1 SA key should be listed after the other one is deleted'
+  ).to.equal(1);
+
+  key2 = credsList3.filter(key => key.name.includes(keyId2));
+  expect(key2.length,
+    'Only the SA key that was not deleted should be listed'
+  ).to.equal(1);
+});
+
+
+Scenario('Get temporary Service Account creds, get object in bucket, delete creds @reqGoogle', async (fence, users, google, files) => {
+  // Test that we do not have access to data anymore after the SA key is deleted
+
+  const googleProject = fence.props.googleProjectA;
+
+  // Setup
+  await fence.complete.forceLinkGoogleAcct(users.user0, googleProject.owner);
```

Comment:
we don't need to link for temp creds access, this is only for User SA data access